### ISSUE TITLE
Program hanging and timing out whilst trying the Marshal results from getManagedCluster call

### ISSUE THREAD

**Issue by pierskarsenbarg** [pulumi/pulumi-azure-native#2620]
### What happened?

Running `azure-native:containerservice:getManagedCluster` function but either the marshaling of the `workloadAutoScalerProfile` property or whatever happens next is timing out. 

Logs are showing the following:

```
2023-07-29T04:08:53.1584601Z     pulumi:pulumi:Stack azure-eastus-azure running I0729 04:08:41.984587    2153 log.go:80] Marshaling property for RPC[azure-native.Invoke(azure-native:containerservice:getManagedCluster).response]: workloadAutoScalerProfile=***map[]***
2023-07-29T05:07:23.5320022Z ##[error]The action has timed out.
```

I have the rest of the logs so you can ping me in Slack.

### Expected Behavior

The program doesn't time out.

### Steps to reproduce

n/a

### Output of `pulumi about`

TBD

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by guineveresaenger** [comment]
Hi @pierskarsenbarg - please feel free to send us your program so we can take a look.

**Comment by pierskarsenbarg** [comment]
I've got the code (but its customer code so can't just upload it). Looking at the section that's calling `get_managed_cluster` there doesn't seem to be anything unusual about it:

```python
aks_cluster_properties = pulumi.Output.all(resource_group.name, managed_cluster.name).apply(
        lambda args: containerservice.get_managed_cluster(
            resource_group_name=args[0],
            resource_name=args[1],
        )
    )
```

**Comment by pierskarsenbarg** [comment]
I've also got a further error message:

I think the key bit is here:

```zsh
AssertionError: Unexpected type. Expected 'list' got 'typing.Mapping[str, typing.Union[str, typing.Awaitable[str], pulumi.output.Output[NoneType]]]'
```

But click "Show all" to see the entire error message:

<details><summary>Show all</summary>

```zsh
Diagnostics:
pulumi:pulumi:Stack (azure-azure-preview):
2023/08/04 16:28:32 [INFO] Creating Provider
clusters/lkc-6oyr23
clusters/lkc-6oyr23
error: Program failed with an unhandled exception:
Traceback (most recent call last):
File "/opt/hostedtoolcache/pulumi/3.73.0/x64/pulumi-language-python-exec", line 197, in <module>
loop.run_until_complete(coro)
File "/usr/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
return future.result()
File "/home/runner/work/iac/iac/pulumi/azure/venv/lib/python3.10/site-packages/pulumi/runtime/stack.py", line 137, in run_in_stack
await run_pulumi_func(lambda: Stack(func))
File "/home/runner/work/iac/iac/pulumi/azure/venv/lib/python3.10/site-packages/pulumi/runtime/stack.py", line 51, in run_pulumi_func
await wait_for_rpcs()
File "/home/runner/work/iac/iac/pulumi/azure/venv/lib/python3.10/site-packages/pulumi/runtime/stack.py", line 83, in wait_for_rpcs
raise exn from cause
File "/home/runner/work/iac/iac/pulumi/azure/venv/lib/python3.10/site-packages/pulumi/runtime/rpc_manager.py", line 71, in rpc_wrapper
result = await rpc
File "/home/runner/work/iac/iac/pulumi/azure/venv/lib/python3.10/site-packages/pulumi/runtime/resource.py", line 872, in do_register
resolver = await prepare_resource(res, ty, custom, remote, props, opts, typ)
File "/home/runner/work/iac/iac/pulumi/azure/venv/lib/python3.10/site-packages/pulumi/runtime/resource.py", line 176, in prepare_resource
serialized_props = await rpc.serialize_properties(
File "/home/runner/work/iac/iac/pulumi/azure/venv/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 209, in serialize_properties
result = await serialize_property(
File "/home/runner/work/iac/iac/pulumi/azure/venv/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 337, in serialize_property
await serialize_property(
File "/home/runner/work/iac/iac/pulumi/azure/venv/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 498, in serialize_property
return {
File "/home/runner/work/iac/iac/pulumi/azure/venv/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 499, in <dictcomp>
k: await serialize_property(
File "/home/runner/work/iac/iac/pulumi/azure/venv/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 333, in serialize_property
element_type = _get_list_element_type(typ)
File "/home/runner/work/iac/iac/pulumi/azure/venv/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 162, in _get_list_element_type
raise AssertionError(f"Unexpected type. Expected 'list' got '{typ}'")
AssertionError: Unexpected type. Expected 'list' got 'typing.Mapping[str, typing.Union[str, typing.Awaitable[str], pulumi.output.Output[NoneType]]]'
exception calling callback for <Future at 0x7fcd14d215a0 state=finished returned RegisterResourceResponse>
Traceback (most recent call last):
File "/usr/lib/python3.10/concurrent/futures/_base.py", line 342, in _invoke_callbacks
callback(self)
File "/usr/lib/python3.10/asyncio/futures.py", line 399, in _call_set_state
dest_loop.call_soon_threadsafe(_set_state, destination, source)
File "/usr/lib/python3.10/asyncio/base_events.py", line 795, in call_soon_threadsafe
self._check_closed()
File "/usr/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
exception calling callback for <Future at 0x7fcd154c7190 state=finished returned RegisterResourceResponse>
Traceback (most recent call last):
File "/usr/lib/python3.10/concurrent/futures/_base.py", line 342, in _invoke_callbacks
callback(self)
File "/usr/lib/python3.10/asyncio/futures.py", line 399, in _call_set_state
dest_loop.call_soon_threadsafe(_set_state, destination, source)
File "/usr/lib/python3.10/asyncio/base_events.py", line 795, in call_soon_threadsafe
self._check_closed()
File "/usr/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
exception calling callback for <Future at 0x7fcd154c7400 state=finished returned RegisterResourceResponse>
Traceback (most recent call last):
File "/usr/lib/python3.10/concurrent/futures/_base.py", line 342, in _invoke_callbacks
callback(self)
File "/usr/lib/python3.10/asyncio/futures.py", line 399, in _call_set_state
dest_loop.call_soon_threadsafe(_set_state, destination, source)
File "/usr/lib/python3.10/asyncio/base_events.py", line 795, in call_soon_threadsafe
self._check_closed()
File "/usr/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
exception calling callback for <Future at 0x7fcd154c7640 state=finished returned RegisterResourceResponse>
Traceback (most recent call last):
File "/usr/lib/python3.10/concurrent/futures/_base.py", line 342, in _invoke_callbacks
callback(self)
File "/usr/lib/python3.10/asyncio/futures.py", line 399, in _call_set_state
dest_loop.call_soon_threadsafe(_set_state, destination, source)
File "/usr/lib/python3.10/asyncio/base_events.py", line 795, in call_soon_threadsafe
self._check_closed()
File "/usr/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
exception calling callback for <Future at 0x7fcd150f5ea0 state=finished returned RegisterResourceResponse>
Traceback (most recent call last):
File "/usr/lib/python3.10/concurrent/futures/_base.py", line 342, in _invoke_callbacks
callback(self)
File "/usr/lib/python3.10/asyncio/futures.py", line 399, in _call_set_state
dest_loop.call_soon_threadsafe(_set_state, destination, source)
File "/usr/lib/python3.10/asyncio/base_events.py", line 795, in call_soon_threadsafe
self._check_closed()
File "/usr/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
Resources: + 35 created
```

</details>

**Comment by pierskarsenbarg** [comment]
Looks like using `list_managed_cluster_user_credentials` instead of `list_managed_cluster_user_credentials_output` fixed the threading issues. I wonder if this is associated with pulumi/pulumi#7663

**Comment by thomas11** [comment]
@pierskarsenbarg should we leave this issue open? Is there anything the Azure Native provider could be doing differently here?

**Comment by pierskarsenbarg** [comment]
Let me see if `list_managed_cluster_user_credentials_output` is working now.

**Comment by danielrbradley** [comment]
This looks similar to an issue which @MitchellGerdisch was looking at recently which actually looked related to the Helm chart component being involved.

It looks like this can be worked around by calling the invoke within an apply rather than calling directly with output arguments.

@pierskarsenbarg let us know if this is still an issue.

Internal reference:
- https://github.com/pulumi/customer-support/issues/1382

**Comment by justinvp** [comment]
I _think_ this is the same issue and for the same customer as https://github.com/pulumi/customer-support/issues/1382. We provided a workaround and the customer is now unblocked.

I've opened https://github.com/pulumi/pulumi/issues/15462 to track resolving the underlying issue. I believe we can likely close this issue as a duplicate of https://github.com/pulumi/pulumi/issues/15462.

**Comment by thomas11** [comment]
Thank you @justinvp!
